### ISSUE TITLE
[QUESTION] Conditionally stopping training mid Epoch?

### ISSUE THREAD

**Issue by osvitashev** [takuseno/d3rlpy#459]
Is there a way to conditionally stop the training process mid epoch? For example when the training loss starts to increase? Or if some other condition is met.
Thanks!

**Comment by osvitashev** [comment]
@takuseno Thanks for the answer.
Some background of why i asked:
I am playing with reinforcement learning for game ai. (Just a toy project for me to get a grip on offline RL)
What i am seeing in my experiments is that once the training loss passes a minimum, the training process diverges very quickly and i am struggling to capture the best achieved parameters.
One solution is to have short epochs with a lot of checkpoints But then my program spends a lot of time doing file I/O that i do not really need.
